test/Core.Test/AdminConsole/Services/PolicyServiceTests.cs

```diff
@@ -341,6 +360,75 @@ await sutProvider.GetDependency<IPolicyRepository>().Received()
         Assert.True(policy.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
     }
 
+    [Theory, BitAutoData]
+    public async Task SaveAsync_ExistingPolicy_UpdateTwoFactor_WithoutMasterPasswordOr2FA_ThrowsBadRequest(
+        [AdminConsoleFixtures.Policy(PolicyType.TwoFactorAuthentication)] Policy policy, SutProvider<PolicyService> sutProvider)
+    {
+        // If the policy that this is updating isn't enabled then do some work now that the current one is enabled
+
+        var org = new Organization
+        {
+            Id = policy.OrganizationId,
+            UsePolicies = true,
+            Name = "TEST",
+        };
+
+        SetupOrg(sutProvider, policy.OrganizationId, org);
+
+        sutProvider.GetDependency<IPolicyRepository>()
+            .GetByIdAsync(policy.Id)
+            .Returns(new Policy
+            {
+                Id = policy.Id,
+                Type = PolicyType.TwoFactorAuthentication,
+                Enabled = false,
+            });
+
+        var orgUserDetailUser = new Core.Models.Data.Organizations.OrganizationUsers.OrganizationUserUserDetails
```

Comment:
nit, seeing as how I'm already asking for changes: please bring this into scope with `using` instead of the super long chain of namespaces.